woocommerce/rest-api/Controllers/Settings.php

```diff
@@ -194,13 +194,47 @@ public function update_item_permissions_check( $request ) {
 	}
 
 
-	// TODO: update_item()
+	/**
+	 * Updates a single setting.
+	 *
+	 * @since x.y.z
+	 *
+	 * @param \WP_REST_Request $request request object
+	 * @return WP_Error|WP_REST_Response
+	 */
+	public function update_item( $request ) {
+
+		try {
+
+			$setting_id = $request->get_param( 'id' );
+			$value      = $request->get_param( 'value' );
+
+			// throws an exception if the setting doesn't exist or the value is not valid
+			$this->settings->update_value( $setting_id, $value );
+
+			return rest_ensure_response( $this->prepare_setting_item( $this->settings->get_setting( $setting_id ), $request ) );
+
+		} catch ( \Exception $e ) {
+
+			return new \WP_Error(
```

Comment:
I'm not sure if that is what @ChaseWiseman meant by `Exception codes are passed to any \WP_Error` when he wrote the acceptance criteria, but should we maybe return different errors for each case (not found or invalid value)? We would have to update the `Abstract_Settings::update_value` and `Setting::update_value` to throw different exceptions. But it would allow the UI using this API to display specific error messages... 🤔

Reply:
@deivamagalhaes you make a good point. The errors would be more useful that way.

You mean changing `wc_rest_setting_could_not_update` to `wc_rest_setting_not_found` and `wc_rest_setting_invalid_value` depending on the type of exception?

If we add `400` and `404` codes to the exception thrown when the value is invalid and when the setting doesn't exist, we may be able to check the code on the `catch` block and adjust the error code and message accordingly.
Do we normally use the exceptions codes that way?

I think I prefer to have two new exceptions for these specific cases, but we also don't have many subclasses of `SV_WC_Payment_Gateway_Exception` either?

What do you think?

Reply:
> You mean changing wc_rest_setting_could_not_update to wc_rest_setting_not_found and wc_rest_setting_invalid_value depending on the type of exception?

Yes, this would be what I would prefer. 

But maybe this point is something we should discuss with the whole team?

Reply:
@deivamagalhaes @wvega sorry for being a bit vague there!

I meant the code we set for the `\WP_Error`'s `'status'` value. That can be set using `$e->getCode()`, and whenever throwing them in the related code we'd pass the appropriate HTTP code depending on the problem. `404` for not found,`400` for invalid, etc... whatever we set there ends up as the HTTP response code.

I think the _string_ error code, `wc_rest_setting_could_not_update` can remain the same either way

Reply:
Thank you @deivamagalhaes @ChaseWiseman.

I updated `Abstract_Settigns::update_value()` and `Setting::update_value()` to add the appropriate error code (`404` or `400`) to the exceptions they throw (9f8cae5 and 269def0) and modified the tests to check those values (1abd1c1).

Also updated the story description with a link to this conversation.